src/containers/FileViewers.tsx

```diff
@@ -0,0 +1,389 @@
+                      </Grid>
+                    );
+                  })}
+                  {apiResults.nextPageLink == null ? (
+                    <></>
+                  ) : (
+                    <Grid item style={{ padding: '0' }}>
+                      <ButtonBase
+                        style={{
+                          width: '100%',
+                          height: '100%',
+                          justifyContent: 'center',
+                          backgroundColor: grey[100],
+                          padding: '0.5rem',
+                        }}
+                        onClick={() => setQueryNextUrlString(apiResults.nextPageLink)}>
+                        Load More
+                      </ButtonBase>
+                    </Grid>
+                  )}
+                </Grid>
+              </Grid>
+              <Grid
+                item
+                container
+                xs={8}
+                style={{
+                  borderLeft: `1pt solid ${grey[300]}`,
+                  height: '100%',
+                  position: 'relative',
+                  backgroundColor: `${grey[500]}`,
+                }}
+                direction='column'
+                justifyContent='center'
+                alignItems='center'>
+                {selectedPreview == null ? (
+                  <CircularProgress />
+                ) : (
+                  <Grid item style={{ maxHeight: '100%', position: 'unset', padding: '1rem' }}>
+                    <div style={{ height: '48px' }}>
+                      <LimsRowDetailsDialog
+                        dialogOpened={isDialogOpen}
+                        rowData={selectedPreview}
+                        onDialogClose={() => setIsDialogOpen(false)}
+                      />
+                      <ImageListItemBar
+                        position='top'
+                        title={selectedPreview.key}
+                        actionIcon={
+                          <IconButton onClick={() => setIsDialogOpen(true)}>
+                            <InfoIcon />
+                          </IconButton>
+                        }
+                      />
+                    </div>
+                    <img
+                      style={{
+                        maxHeight: 'calc(100% - 48px)',
+                        maxWidth: '100%',
+                        backgroundColor: 'white',
+                        padding: '1px',
+                      }}
+                      onClick={() => window.open(selectedPreview.presigned_url, '_blank')}
+                      src={selectedPreview.presigned_url}
+                    />
+                  </Grid>
+                )}
+              </Grid>
+            </Grid>
+          )}
+        </Grid>
+      </Grid>
+    </Paper>
+  );
+}
+
+async function getPresignedUrlForEveryItem(apiResponse: any) {
+  for (const item of apiResponse.results) {
+    const id = item.id;
+    const presigned_url = await getPreSignedUrl(id);
```

Comment:
Can we lazy fetch the PreSignedUrl request here, pls? i.e. Just grab one for page entry rendering purpose. Then, only when, user started clicking around next item then make a request and, push into local state for caching/reusing, for example.

Reply:
And, pls use `CircularProgress` or appropriate component for transition effect, if any.

Reply:
(Hehe, don't hate me! LoL) If it gets tedious to do, let me know. We can try another option/idea. @williamputraintan

Reply:
> And, pls use CircularProgress or appropriate component for transition effect, if any.

When using this function, it is sandwiched with [isLoading state ](https://github.com/umccr/data-portal-client/blob/fa7089df0910417980e3d5795a89919b6a3c2817/src/containers/FileViewers.tsx#L92)that have a CircularProgress component.

Reply:
>  don't hate me

Nope, not hating you! 😅 _(It should be this way in the beginning)_

> We can try another option/idea.

Did another push to resolve this 😉

Reply:
Great! I'll check out in tick.